### ISSUE TITLE
[Project Addition]: Sentiment Analysis for Restaurant Reviews

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#618]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Sentiment Analysis for Restaurant Reviews
:red_circle: **Aim** : The aim is to analyze the reviews collected in the dataset.
:red_circle: **Dataset** : https://www.kaggle.com/datasets/d4rklucif3r/restaurant-reviews
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by SiMi723** [comment]
Full name : Simi
GitHub Profile Link : https://github.com/SiMi723
Participant ID (If not, then put NA) : NA
Approach for this Project :
What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.):Contributor(SSOC)

**Comment by pratikringe46** [comment]
Hey @abhisheks008,

Can you please assign me this issue under SSOC season 3?
Full Name: Pratik Ringe
[Github](https://github.com/pratikringe46)
Participation ID: NA
Participant Role: SSOC season 3
My approach:
I will be trying 3-4 algos for this: Logistic regression, Naive bayes, SVM, Neural Networks. I have worked on classification and regression models before. The idea would be to implement these model and also provide a comparison between them based on the accuracy and other metrics. I can try using LSTM as well.

Thanks.

**Comment by aryamanpathak2022** [comment]
Hello @abhisheks008 

Full name: [Aryaman Pathak]
GitHub Profile Link: [https://github.com/aryamanpathak2022]
Participant ID: NA
Approach for this Project:
I am excited to contribute to the Sentiment Analysis for Restaurant Reviews project. My approach will include:
1. **Exploratory Data Analysis (EDA)**: Understanding the dataset through visualizations and summary statistics.
2. **Data Preprocessing**: Cleaning and preparing the data for analysis, including tokenization, removing stopwords, and other text preprocessing techniques.
3. **Model Implementation**: Implementing 3-4 machine learning algorithms such as Logistic Regression, Random Forest, SVM, and Naive Bayes for sentiment analysis.
4. **Model Comparison**: Comparing the performance of the models using accuracy scores and other relevant metrics to determine the best-fit model.
5. **Documentation**: Documenting the entire process, including the EDA, preprocessing steps, model implementations, comparisons, and conclusions in the README.md file.

What is your participant role?: SSOC (Social Summer of Code)

Additional Information:
I am currently working on a similar sentiment analysis project focused on Western news about India, which has given me relevant experience and knowledge. You can view my ongoing project here: [https://github.com/aryamanpathak2022/Sentimental_analysis_wn ]

Happy Contributing 🚀

**Comment by keshav1441** [comment]
Full name : Keshav Sharma
GitHub Profile Link : https://github.com/keshav1441
Participant ID : NA
Approach for this Project :
My approach towards implementing sentiment analysis for restaurant reviews is , first, collect a dataset of labeled restaurant reviews. Preprocess the text by tokenizing, removing stop words, and normalizing. Use a machine learning model like logistic regression, or a deep learning model like LSTM, to train on the dataset. Finally, evaluate the model's performance using metrics such as accuracy, precision, recall, and F1-score.
Participant Role : contributor

**Comment by abhisheks008** [comment]
> Hello Sir @abhisheks008 Full name : Simi GitHub Profile Link : https://github.com/SiMi723 Participant ID (If not, then put NA) : NA Approach for this Project :Implement at least 3-4 different algorithms such as: Logistic Regression Support Vector Machine (SVM) Random Forest Naive Bayes Deep Learning models (e.g., LSTM, BERT) Train and validate each model using appropriate metrics.Use an appropriate algorithm accordingly. What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.):Contributor(SSOC & GSSoC) Sir,i am really excited to learn the algorithm of machine learning and looking forward to contribute in this project.

Issue assigned to you @SiMi723 

Make sure you implement all these models,
1. Random Forest
2. Decision Tree
3. Logistic Regression
4. Gradient Boosting
5. XGBoost
6. Lasso
7. Ridge
8. MLP Classifier

**Comment by SiMi723** [comment]
@abhisheks008 Thank you Sir for giving me this opportunity .I will try to implement the above mentioned models.

**Comment by akashinferno** [comment]
Full name : Akash Vardhan V
GitHub Profile Link : https://github.com/akashinferno
Participant ID  : NA
Approach for this Project :  Try to use 4-5 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also  a exploratory data analysis before creating any model.
What is your participant role? : GITRECQUEST 
@abhisheks008
